lib/segment/src/vector_storage/quantized/quantized_vectors.rs

```diff
@@ -1009,6 +1084,266 @@ impl QuantizedVectors {
+            }
+            _ => {
+                debug_assert!(
+                    false,
+                    "Only binary quantization is supported for mutable quantized vectors"
+                );
+                Ok(None)
+            }
+        }
+    }
+
+    fn create_mutable_binary(
+        binary_config: BinaryQuantizationConfig,
+        config: QuantizedVectorsConfig,
+        vector_storage: &VectorStorageEnum,
+        path: &Path,
+    ) -> OperationResult<Self> {
+        match vector_storage.try_multi_vector_config() {
+            Some(multi_vector_config) => Self::create_mutable_binary_multi(
+                binary_config,
+                config,
+                *multi_vector_config,
+                vector_storage,
+                path,
+            ),
+            None => Self::create_mutable_binary_dense(binary_config, config, vector_storage, path),
+        }
+    }
+
+    fn create_mutable_binary_dense(
+        binary_config: BinaryQuantizationConfig,
+        config: QuantizedVectorsConfig,
+        vector_storage: &VectorStorageEnum,
+        path: &Path,
+    ) -> OperationResult<Self> {
+        let meta_path = Self::get_meta_path(path);
+        let distance = vector_storage.distance();
+        let datatype = vector_storage.datatype();
+        let encoding = Self::convert_binary_encoding(binary_config.encoding);
+        let query_encoding = Self::convert_binary_query_encoding(binary_config.query_encoding);
+
+        let quantization_storage = Self::load_mutable_binary_storage::<u128>(
+            &binary_config,
+            &config,
+            vector_storage,
+            path,
+        )?;
+        let storage_impl =
+            QuantizedVectorStorage::BinaryChunkedMmap(EncodedVectorsBin::new_empty_appendable(
+                quantization_storage,
+                config.vector_parameters.clone(),
+                encoding,
+                query_encoding,
+                meta_path.as_path(),
+            )?);
+        Ok(QuantizedVectors {
+            storage_impl,
+            config,
+            path: path.to_path_buf(),
+            distance,
+            datatype,
+        })
+    }
+
+    fn create_mutable_binary_multi(
+        binary_config: BinaryQuantizationConfig,
+        config: QuantizedVectorsConfig,
+        multi_vector_config: MultiVectorConfig,
+        vector_storage: &VectorStorageEnum,
+        path: &Path,
+    ) -> OperationResult<Self> {
+        let meta_path = Self::get_meta_path(path);
+        let distance = vector_storage.distance();
+        let datatype = vector_storage.datatype();
+        let encoding = Self::convert_binary_encoding(binary_config.encoding);
+        let query_encoding = Self::convert_binary_query_encoding(binary_config.query_encoding);
+
+        let quantization_storage =
+            Self::load_mutable_binary_storage::<u8>(&binary_config, &config, vector_storage, path)?;
```

Comment:
Why do we load binary storage with generic type <u8> here and <u128> in the function above?
Can we maybe use a type alias instead to make it more clear what these sizes mean and why they've been picked?